GLOSSARY.md

```diff
@@ -0,0 +1,299 @@
+# Glossary
+
+This is a glossary of terms that will be commonly used in discussion. This list
+will be expanded in the future. **Please avoid changing existing headings as
+users may be deep-linking to the generated anchors.**
+
+## Table of Contents
+
+* [Error Symposium](#error-symposium)
+  * [Operational Error](#operational-error)
+  * [Programmer Error](#programmer-error)
+  * [Expected versus Unexpected](#expected-versus-unexpected)
+* [Post Mortem Analysis](#post-mortem-analysis)
+  * [Abort](#abort)
+  * [Core / Core dump / Core file](#core--core-dump--core-file)
+  * [Stack Frame](#stack-frame)
+  * [Unwinding the Stack / Top of Stack](#unwinding-the-stack--top-of-stack)
+* [Promise](#promise)
+  * [Executor / Resolver](#executor--resolver)
+  * [Handler](#handler)
+  * [Settling](#settling)
+    * [Rejection / reject](#rejection--reject)
+    * [Resolving / resolve](#resolving--resolve)
```

Comment:
I find it useful to organize this as:

- promise states: pending or settled
- settled states: fulfilled or rejected

- defer (to create a pending promise, or to resolve a pending promise with another pending promise)
- resolve (to take a pending promise and, having made some progress, to designate another promise (albeit a fulfilled, rejected, or pending promise) as its resolution, responsible for eventually notifying its observers when it is ultimately settled)
- fulfill (to create a fulfilled promise, or resolve and settle a pending promise with a fulfilled promise, eventually notifying all of its eventual handlers of the fulfillment value)
- reject (to create a rejected promise, or resolve and settle a pending promise with a rejected promise, eventually notifying all of its eventual handlers of the rejection error)